packages/server/src/sdk/app/automations/crud.ts

```diff
@@ -281,4 +285,14 @@ function guardInvalidUpdatesAndThrow(
       }
     })
   }
+
+  if (isRowAction(automation) && automation.name !== oldAutomation.name) {
+    throw new Error("Row actions cannot be renamed")
+  }
+}
+
+function isRowAction(automation: Automation) {
+  const result =
+    automation.definition.trigger.stepId === AutomationTriggerStepId.ROW_ACTION
+  return result
```

Comment:
Should we throw this into shared-core so we can use it on the frontend too?

Reply:
Good call!